beta/src/components/Layout/Page.tsx

```diff
@@ -15,13 +15,25 @@ interface PageProps {
 }
 
 export function Page({routeTree, children}: PageProps) {
+  const [showBanner, setShowBanner] = React.useState(true);
+  const updateVisibility = () => setShowBanner(window.scrollY <= 6);
```

Comment:
It would be nice to think of some way to not run `set*` function on every single scroll event (even if it bails out). Could we do that?

Reply:
Yes, let me change it